source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,281 @@
+</blockquote>
+
+### `typeDefs` (optional)
+
+GraphQL type definitions provided by this module. As a developer, this is a contract with the rest of your codebase.
+
+Type definitions must [GraphQL schema language strings](https://graphql.org/learn/schema/), and can be declared inline or loaded from a `gql` file. One can create new types, add top level queries (by extending the `Query` type), mutations or extend types from other modules.
+
+```js
+// or import typeDefs from './schema.gql';
+const typeDefs = `
+extend type Query {
+  "Be polite!"
+  sayHello (name: String!): Message
+}
+
+type Message {
+  response: String
+  audio: String
+}
+`
+
+module.exports = {
+  namespace: "Acme/HelloWorld",
+  typeDefs: typeDefs
+};
+```
+
+<blockquote class="info">
+**In depth:** under the hood, all module’s `typeDefs` are merged in a single array passed to graphql-tools' [`makeExecutableSchema`](https://www.apollographql.com/docs/graphql-tools/generate-schema.html#makeExecutableSchema) function
+</blockquote>
+
+### `resolvers` (optional)
+
+This is an object containing all the resolvers for resolving fields defined in the schema (usually from your `typeDefs`). This is where the implementation resides.
+
+Resolver map must follow [the format documented in Apollo Tools](https://www.apollographql.com/docs/graphql-tools/resolvers.html).
+
+```js
+// or import resolvers from './resolvers.js';
+const resolvers = {
+  Query: {
+    sayHello: (_, { name }) => {
+      const message = `Hello ${name}`;
+      return {
+        message: message,
+        audio: `https://tts.service.com/q=${message}`
+      };
+    }
+  }
+};
+
+module.exports = {
+  namespace: "Acme/HelloWorld",
+  // …
+  resolvers: resolvers
+};
+```
+
+<blockquote class="info">
+**In depth:** under the hood, all module’s `resolvers` are merged in a single object (using [lodash.merge](https://lodash.com/docs/#merge) passed to graphql-tools' [`makeExecutableSchema`](https://www.apollographql.com/docs/graphql-tools/generate-schema.html#makeExecutableSchema) function. It is possible to override resolvers declared in another module by declaring it as a dependency and registering a local resolver.
+</blockquote>
+
+### `contextEnhancer` (optional)
+
+The `contextEnhancer` module definition key should be a function that initializes code that will be made available in GraphQL’s context, under the `loaders` key. **It is specific to Front-Commerce.**
+
+This is where you could construct models or [dataloaders](#TODO) that should be used in your GraphQL resolvers.
+
+The `contextEnhancer` function must return an object whose keys will be merged with previous modules’. It will be passed a single argument with the following keys:
+
+* `req`: the current server request
+* `loaders`: current loaders (from module initialized beforehand)
+* `makeDataLoader`: a factory to build a dataloader (see [dataloaders](#TODO))
+* `config`: the global configuration
```

Comment:
It's a bit weird to talk about `globalConfiguration` because I would expect to have access to files in `src/config` for instance. Maybe we could at least talk about `the server configuration`?